python_modules/dagster/dagster_tests/definitions_tests/declarative_automation_tests/scenario_utils/asset_daemon_scenario.py

```diff
@@ -212,6 +212,7 @@ def _evaluate_tick_daemon(
                     threadpool_executor=self.threadpool_executor,
                     amp_tick_futures=amp_tick_futures,
                     debug_crash_flags={},
+                    submit_threadpool_executor=None,
```

Comment:
i don't know that we need to be running all these scenarios with the submit threadpool executor at all times, but at least as a one-off i think it would be useful to do it once manually to suss out any potential problems

Reply:
Is the concern here that the behavior of AMPs might differ in some way from the behavior of AutomationConditions?

Generally, I want to move away from this "scenario" pattern, as I've found that the `test_e2e.py` (and the new test file I added in this PR) are a better representation of "real world" conditions. So to that end, I did update one of the e2e tests to use a submit threadpool executor. 

Down to add an e2e test that uses AMPs, just want to avoid extending the scenario framework at this point, as my eventual goal is to delete all of it